acceptance/tests/facts/macosx.rb

```diff
@@ -0,0 +1,97 @@
+test_name "Facts should resolve as expected in Mac OS X 10.9 and 10.10"
+
+#
+# This test is intended to ensure that facts specific to an OS configuration
+# resolve as expected in Mac OS X 10.9 and 10.10
+#
+# Facts tested: os, processors, networking, identity, kernel
+#
+
+confine :to, :platform => /osx-mavericks|osx-yosemite/
+
+agents.each do |agent|
+  if agent['platform'] =~ /osx-mavericks/
+    os_version = '10.9'
+    kernel_major = '13'
+  else
+    os_version = '10.10'
+    kernel_major = '14'
+  end
+
+  step "Ensure the OS fact resolves as expected"
+
+  expected_os = {
+                  'os.architecture'         => 'x86_64',
+                  'os.family'               => 'Darwin',
+                  'os.hardware'             => 'x86_64',
+                  'os.name'                 => 'Darwin',
+                  'os.macosx.build'         => /#{kernel_major}[A-F]\d+/,
```

Comment:
I didn't find a specification, but I think that letter doesn't have to be in the "hex range" of A-F. E.g. see some of the build versions here: https://support.apple.com/en-us/HT204319